src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -263,32 +257,24 @@ public void ClearMarkers(ScriptFile file)
         /// <param name="settings">The new language server settings.</param>
         public void OnConfigurationUpdated(object sender, LanguageServerSettings settings)
         {
-            InitializeAnalysisEngineToCurrentSettings();
+            if (settings.ScriptAnalysis.Enable ?? true)
+            {
+                InitializeAnalysisEngineToCurrentSettings();
+            }
```

Comment:
I'm not sure on this... I think this has to run `InitializeAnalysisEngineToCurrentSettings` every time for it to be accurate.

Reply:
I don't think this is a concern.